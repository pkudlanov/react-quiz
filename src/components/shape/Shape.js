import React from 'react';
import PropTypes from 'prop-types';

function Shape({
  text,
  color,
  backgroundColor
}){
  return(
    <p style={{ color, backgroundColor }}>
      {text}
    </p>
  );
}

Shape.propTypes = {
  color: PropTypes.string.isRequired,
  backgroundColor: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired
};

export default Shape;
