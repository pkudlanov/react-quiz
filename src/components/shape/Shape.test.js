import React from 'react';
import { shallow } from 'enzyme';
import Shape from './Shape';

describe('Shape component', () => {
  it('renders Shape', () => {
    const wrapper = shallow(<Shape
      color="green"
      backgroundColor="orange"
      text="Greetings friends and foe!!!!" />);
    expect(wrapper).toMatchSnapshot();
  });
});
