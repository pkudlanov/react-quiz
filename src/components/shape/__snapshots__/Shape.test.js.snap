// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shape component renders Shape 1`] = `
<p
  style={
    Object {
      "backgroundColor": "orange",
      "color": "green",
    }
  }
>
  Greetings friends and foe!!!!
</p>
`;
